src/appkit.cs

```diff
@@ -589,8 +589,13 @@ interface NSApplication : NSAccessibilityElementProtocol, NSUserInterfaceValidat
 		[Availability (Deprecated = Platform.Mac_10_12, Message = "Use EnumerateWindows instead.")]
 		NSWindow MakeWindowsPerform (Selector aSelector, bool inOrder);
 	
+		[Deprecated (PlatformName.MacOSX, 10, 7, message : "Remove usage or use 'DangerousWindows' instead.")]
+		[EditorBrowsable (EditorBrowsableState.Never)]
 		[Export ("windows")]
 		NSWindow [] Windows { get; }
+
+		[Advice ("Use of DangerousWindows can prevent windows from leaving memory.")]
+		NSWindow [] DangerousWindows { [Wrap ("Windows")] get; }
```

Comment:
why not expose a `NSArray<NSWindow>` ? which would be _safe(r)_ to use ?

Reply:
This is actually not a bad idea, this would require the `[Export ("windows")]` be on `DangerousWindows` instead and Wrap would become `NSWindow [] Windows { get; }` just make sure `IsVirtual = true` on `Wrap`

Reply:
yes :) but it also does not have to be named `Dangerous*` since (from what I read in the original issue) I think it's the managed array that caused the _retain_ problem.

Reply:
So I think making it an NSArray will make the problem better but not completely gone. 

The windows property hooks into dealloc to remote itself from the list, so if you have a window that you closed, but is not dealloced, and then reference the windows and walk the array, you'll resurrect that closed window. If you walk the list often enough (which customers were doing), it'll stay around forever.

Reply:
Let's keep the Danger around :)

Reply:
![beautiful-i-also-like-to-live-dangerously-meme-if-you-are-on-gateway-what-launcher-gbatemp-the-i-also-like-to-live-dangerously-meme](https://user-images.githubusercontent.com/204671/53265695-e245a280-36ac-11e9-95d3-26d04b5f814d.jpg)